test/data-driven/text-fragments-data-driven-test.js

```diff
@@ -0,0 +1,239 @@
+        directive,
+        )['text'];
+
+    if (processedDirectives.count == 0) {
+      return false;
+    }
+    const marks = processedDirectives[0];
+    const highlightingTextMatch =
+        marksArrayToString(marks) == testInput.highlightText;
+
+    return highlightingTextMatch;
+  }
+
+  /**
+   * Generates the output of a test case.
+   * @param {Boolean} generationSuccess - True if fragment generation test was
+   *     successful.
+   * @param {Boolean} highlightingSuccess - True if highlighting test was
+   *     successful.
+   * @return {String} The test's output text that will be compared with the
+   *     expected output in the test's output file.
+   */
+  generateOutput(generationSuccess, highlightingSuccess) {
+    return `GENERATION: ${this.getTextForTestResult(generationSuccess)}\n` +
+        `HIGHLIGHTING: ${this.getTextForTestResult(highlightingSuccess)}\n`
+  }
+
+  /**
+   * Helper method to get the textual representation of a test result that is
+   * included in the data driven test output.
+   * @param {Boolean} testSuccess - True if test was successful.
+   * @return {String} {@link OUTPUT_SUCCESS} when |testSuccess| is true
+   *     or {@link OUTPUT_FAILURE} when false.
+   */
+  getTextForTestResult(testSuccess) {
+    return testSuccess ? OUTPUT_SUCCESS : OUTPUT_FAILURE;
+  }
+
+  /**
+   * Text Fragments Data Driven Test input object.
+   * @typedef {Object} TextFragmentDataDrivenTestInput
+   * @property {String} htmlFileName - Html file name for the test case. This file should be in |htmlDirectory|.
+   * @property {String} [startParentId] - Id of the parent of the node where the
+   * test case’s selection starts. Optional: when null then the parent node is
+   * the document's body.
+   * @property {Number} startOffsetInParent - Offset of the selection’s starting
+   * node in its parent node.
+   * @property {Number} [startTextOffset] - When the selection starts in a text
+   * node, this value indicates the offset inside the node’s text where the
+   * selection starts.
+   * @property {String} [endParentId] - Id of the parent of the node where the
+   * test case’s selection ends. Optional: when null then the parent node is the
+   * document's body.
+   * @property {Number} endOffsetInParent - Offset of the selection’s ending
+   * node in its parent node.
+   * @property {Number} [endTextOffset] - When the selection ends in a text
+   * node, this value indicates the offset inside the node’s text where the
+   * selection ends.
+   * @property {String} selectedText - Expected text content inside the
+   * selection.
+   * @property {String} highlightText - Expected text content inside the range
+   * highlighted.
+   */
+
+  /**
+   * Parses the content of a test case input file.
+   * @param {String} input - Content of a test case input file.
+   * @return {TextFragmentDataDrivenTestInput} Test Case input object.
+   */
+  parseInput(input) {
+    return JSON.parse(input);
```

Comment:
Should this be wrapped in a try/catch block in case of malformed JSON? That way we could give a better error message (I seem to recall JSON parse error messages are not always super clear)